runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/WriteFilesTranslation.java

```diff
@@ -53,28 +59,25 @@
   public static final String CUSTOM_JAVA_FILE_BASED_SINK_URN =
       "urn:beam:file_based_sink:javasdk:0.1";
 
-  public static final String CUSTOM_JAVA_FILE_BASED_SINK_FORMAT_FUNCTION_URN =
-      "urn:beam:file_based_sink_format_function:javasdk:0.1";
-
   @VisibleForTesting
   static WriteFilesPayload toProto(WriteFiles<?, ?, ?> transform) {
+    Map<String, SideInput> sideInputs = Maps.newHashMap();
+    for (PCollectionView<?> view : transform.getSink().getDynamicDestinations().getSideInputs()) {
+      sideInputs.put(view.getTagInternal().getId(), ParDoTranslation.toProto(view));
```

Comment:
`WriteFilesTranslation` should not depend on `ParDoTranslation`. This method made public should be factored into `PCollectionViewTranslation`.

It should have heavy caveats, since it is not correct by construction, but only by being called in a fairly particular way. That is why it was private.